tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -520,7 +520,11 @@ private void OnRoutingEndpointMatched(object arg)
                     tags.AspNetCoreEndpoint = routeEndpoint.Value.DisplayName;
                 }
 
-                var routePattern = routeEndpoint.Value.RoutePattern;
+                if (!routeEndpoint.Value.RoutePattern.TryDuckCast<RoutePattern>(out var routePattern))
+                {
+                    // Failed to cast
```

Comment:
Should we add an error message if failed to have visibility? Is this cast fail expected in any scenario?
routeEndpoint.Value.RoutePattern should already be a RoutePattern. Is this TryDuckCast needed?

Reply:
I don't see where this can fail in any scenario. I changed it this way with a possible error if this case could happen. But to keep the logic the same I changed in b0c83696ce7bda74abf5b42581652b923bdff45f with a simple DuckCast